Add documenation for how to run functional tests.

kdelee opened the pull request:
Includes links to cloudigrade documentation and enumerates required and
optional environment variables for configuring integrade and where to get
necessary information.

Includes make target for running tests against cloudigrade.

Reviewer should verify all items in #9 are satisfied and close the issue.

kdelee committed:
Add documenation for how to run functional tests.

Includes links to cloudigrade documentation and enumerates required and
optional environment variables for configuring integrade and where to get
necessary information.

Includes make target for running tests against cloudigrade.

elyezer commented on Makefile:
```diff
@@ -24,6 +26,9 @@ lint:
 test:
 	py.test tests
 
+test-cloudigrade:
```
Missing an entry on the help target for this new target.

elyezer commented on README.rst:
```diff
@@ -75,8 +75,46 @@ that the tests pass and also the code follows the code style properly. The
+
+    CLOUDIGRADE_API_VERSION # defaults to 'v1'
+    CLOUDIGRADE_USER        # defaults to 'admin'
+    CLOUDIGRADE_USER_PASS   # at the moment we only use the token to
```
Is this already implemented in code? Maybe we can add this when the feature is available.

elyezer commented on README.rst:
```diff
@@ -75,8 +75,46 @@ that the tests pass and also the code follows the code style properly. The
+    CLOUDIGRADE_USER_PASS   # at the moment we only use the token to
+                            # authenticate, but this will be useful later
+    USE_HTTPS  # defaults to false so communication is done over http.
+               #  Set to True to use https.
```
SSL_VERIFY below uses false all lower case and this use True with capitalized word. I think we could consistent and maybe use both lowercase since True and False are Python boolean objs.

elyezer replied:
Yeah, the comment was more in the sense of being consistent and said about using lower case in the examples because it is easier to type

elyezer commented on README.rst:
```diff
@@ -75,8 +75,46 @@ that the tests pass and also the code follows the code style properly. The
+OpenShift cluster, you can collect the necessary `CLOUDIGRADE_BASE_URL` by
+inspecting the output of `oc status`.  To create the necessary super user and
+retreive a token for that user, use the cloudigrade make targets `make oc-user`
+and `make oc-user-authenticate`. Then export these values to
```
If those make targets returns the string to be added to the env vars then maybe we could suggest to run these commands:

```
$ export CLOUDGRADE_USER="$(make oc-user)"
$ export CLOUDIGRADE_TOKEN="$(make oc-user-authenticate)"
```

kdelee replied:
`make oc-user` is interactive and prompts you to make a user. `make oc-user-authenticate` is also interactive and asks for the username of the super user you created with `make oc-user`.

I've made a script that will set these values if you source it. Could not figure out how to make it work from within the Makefile. About to push change so you can call `source scripts/oc-auth.sh` to set the variables.

elyezer replied:
I am not sure if make will be able to export those variables. It will be run by a child process of the shell so any exported variables won't got upwards only for children processes.

elyezer commented on README.rst:
```diff
@@ -75,8 +75,46 @@ that the tests pass and also the code follows the code style properly. The
+inspecting the output of `oc status`.  To create the necessary super user and
+retreive a token for that user, use the cloudigrade make targets `make oc-user`
+and `make oc-user-authenticate`. Then export these values to
+`CLOUDIGRADE_TOKEN` and `CLOUDIGRADE_USER`.
```
I assume CLOUDIGRADE_TOKEN is get by the `make oc-user-authenticate`, is that right? If yes I think the order these vars are shown could be exchanged and maybe use respectively to be more clear about what should be assigned to what.

elyezer reviewed (changes requested):
Please update based on the comments. Thank you

kdelee committed:
Add script to create super user and get token.

Adds a script to interact with cloudigrade when served on openshift to create a
super user and set CLOUDIGRADE_USER and CLOUDIGRADE_TOKEN in the shell when the
script is sourced.

kdelee commented:
@elyezer back to you :smile_cat:
